src/main/java/org/opensrp/service/OrganizationService.java

```diff
@@ -184,16 +190,17 @@ public void assignLocationAndPlan(String identifier, String jurisdictionId, Stri
 	 *
 	 * @return the assigned locations and plans
 	 */
-	public List<AssignedLocations> findAssignedLocationsAndPlansByPlanIdentifier(String planIdentifier) {
-		if (StringUtils.isBlank(planIdentifier))
+	public List<AssignedLocations> findAssignedLocationsAndPlansByPlanIdentifier(AssignedLocationAndPlanSearchBean assignedLocationAndPlanSearchBean) {
+		if (assignedLocationAndPlanSearchBean != null && StringUtils.isBlank(assignedLocationAndPlanSearchBean.getPlanIdentifier()))
 			throw new IllegalArgumentException("PlanIdentifier Identifier not specified");
 
-		Long planId = planRepository.retrievePrimaryKey(planIdentifier);
+		Long planId = planRepository.retrievePrimaryKey(assignedLocationAndPlanSearchBean.getPlanIdentifier());
 
         if (planId == null)
             throw new IllegalArgumentException("Plan not found");
 
-		return organizationRepository.findAssignedLocationsByPlanId(planId);
+        assignedLocationAndPlanSearchBean.setPlanId(planId);
+		return organizationRepository.findAssignedLocationsByPlanId(assignedLocationAndPlanSearchBean);
 
 	}
```

Comment:
To make it easy to understand this method, can we retain the original contract and just add the pagination params. You can package the params back when invoking the repository

Reply:
Okay, noted.